crates/polars-plan/src/logical_plan/hive.rs

```diff
@@ -15,92 +15,68 @@ pub struct HivePartitions {
     stats: BatchStats,
 }
 
-#[cfg(target_os = "windows")]
-fn separator(url: &Path) -> char {
-    if polars_io::is_cloud_url(url) {
-        '/'
-    } else {
-        '\\'
-    }
-}
-
-#[cfg(not(target_os = "windows"))]
-fn separator(_url: &Path) -> char {
-    '/'
-}
-
 impl HivePartitions {
-    pub fn get_statistics(&self) -> &BatchStats {
-        &self.stats
+    /// Constructs a new [`HivePartitions`] from a schema reference.
+    pub fn from_schema_ref(schema: SchemaRef) -> Self {
+        let column_stats = schema.iter_fields().map(ColumnStats::from_field).collect();
+        let stats = BatchStats::new(schema, column_stats, None);
+        Self { stats }
     }
 
-    /// Parse a url and optionally return HivePartitions
-    pub(crate) fn parse_url(url: &Path) -> Option<Self> {
-        let sep = separator(url);
-
-        let url_string = url.display().to_string();
+    /// Constructs a new [`HivePartitions`] from a path.
+    ///
+    /// Returns `None` if the path does not contain any Hive partitions.
+    /// Returns `Err` if the Hive partitions cannot be parsed correctly or do not match the given
+    /// [`Schema`].
+    pub fn try_from_path(path: &Path, schema: Option<SchemaRef>) -> PolarsResult<Option<Self>> {
```

Comment:
The biggest change is that this now returns a `PolarsResult<Option<Self>>` rather than `Option<Self>`. This indicates that the operation may fail for various reasons (returns Err) or may not contain any hive partitions at all (returns None).